### PR TITLE
feat(mechanics): absolute velocity and angle options for effects

### PR DESCRIPTION
**Pull request by Hurleveur**
**Feature:** This PR implements the feature request made by X-27 on the discord (it's a quick one)

## Feature Details
Pretty simple : if you add "absolute angle" to an effect, it's angle will not depend on the previous angle of the projectile, just on what you put in it's txt. In the same way, "absolute velocity" means the velocity will not be inherited from the parent.

## UI Screenshots
N/A

## Usage Examples
If you want a grid fire effect, it can now be a real grid. I don't really know what other things this will achieve but it was asked and it's some really simple coding.

## Testing Done
Used a gun with a grid effect with no random angle or velocity to shoot at targets, the grid was perfectly aligned with my screen all the time.

## Performance Impact
N/A

### CODE REVIEW

**Comment by petervdmeer** [comment]
The feature looks useful, but if we have absolute velocities and absolute angles, then I would still like to be able to set the actual absolute velocity and absolute angle (and not just the boolean that tells us those are absolute).

This could be relevant for a weapon that has an effect that needs to show both horizontally and vertically; instead of creating the effect twice (once for each direction) we would be able to specify the effect twice, with two different angles.

**Comment by Hurleveur** [comment]
> The feature looks useful, but if we have absolute velocities and absolute angles, then I would still like to be able to set the actual absolute velocity and absolute angle (and not just the boolean that tells us those are absolute).
> 
> This could be relevant for a weapon that has an effect that needs to show both horizontally and vertically; instead of creating the effect twice (once for each direction) we would be able to specify the effect twice, with two different angles.

Sorry for only reacting now, but it's done
Damn I wish github would show new comments since last time PR was seen in the online interface

File: source/Effect.cpp [hunk 1]
```diff
@@ -58,6 +58,10 @@ void Effect::Load(const DataNode &node)
 			randomSpin = child.Value(1);
 		else if(child.Token(0) == "random frame rate" && child.Size() >= 2)
 			randomFrameRate = child.Value(1);
+		else if(child.Token(0) == "absolute angle" && child.Size() >= 2)
+			absoluteAngle = child.Value(1);
+		else if(child.Token(0) == "absolute velocity" && child.Size() >= 3)
+			absoluteVelocity = new Point(child.Value(1), child.Value(2));
```
Comment (quyykk) [inline source/Effect.cpp:64]: Don't use `new`! That's a memory leak waiting to happen. You can create a new data member, `hasAbsoluteVelocity` to check whether an absolute velocity was specified.

File: source/Effect.h [hunk 2]
```diff
@@ -57,6 +58,10 @@ class Effect : public Body {
 	double randomAngle = 0.;
 	double randomSpin = 0.;
 	double randomFrameRate = 0.;
+	// Absolute means it does not depend from the angle of the projectile before, and can be specified.
```
Comment (Amazinite) [inline source/Effect.h:61]:
```suggestion
	// Absolute values are independent of the parent Body if specified.
```

**Review by Amazinite — CHANGES_REQUESTED** [review]

**Review by Amazinite — APPROVED** [review]

**Comment by Amazinite** [comment]
The creating effects wiki has been updated with these new effect attributes.

**Pull request merged**